core/src/main/java/org/apache/iceberg/PropertiesUpdate.java

```diff
@@ -53,18 +55,34 @@ public UpdateProperties set(String key, String value) {
     Preconditions.checkNotNull(value, "Value cannot be null");
     Preconditions.checkArgument(!removals.contains(key),
         "Cannot remove and update the same key: %s", key);
+    Preconditions.checkArgument(!transforms.containsKey(key),
+        "Cannot transform and update the same key: %s", key);
 
     updates.put(key, value);
 
     return this;
   }
 
+  @Override
+  public UpdateProperties transform(String key, Function<String, String> transformFunc) {
+    Preconditions.checkNotNull(key, "Key cannot be null");
+    Preconditions.checkNotNull(transformFunc, "Transform function cannot be null");
+    Preconditions.checkArgument(!updates.containsKey(key),
+        "Cannot updates and transform the same key: %s", key);
```

Comment:
Nit: should be "update" rather than "updates".

Reply:
Fixed.

Reply:
@rdblue @kbendick can you take a look at this?